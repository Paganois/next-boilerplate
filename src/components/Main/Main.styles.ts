import styled from "styled-components";

export const Wrapper = styled.main`
    background-color: #1e0b36;
    color: #fff;
    width: 100%;
    height: 100%;
    padding: 3rem;

    display: flex;
    flex-direction: column;
    text-align: center;
    align-items: center;
    justify-content: center;
`;

export const Logo = styled.div`
    margin-bottom: 2rem;
`;

export const Title = styled.h1`
    font-size: 2.5rem;
`;

export const Description = styled.h2`
    font-size: 2rem;
    font-weight: 400;
    margin-top: 1rem;
    width: min(50rem, 100%);
`;
