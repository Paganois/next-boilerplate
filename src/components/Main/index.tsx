import * as S from "./Main.styles";

const Main = ({
    title = "Onyx Boilerplate",
    description = "Boilerplate with Typescript, NextJS, Styled Components, PWA, Storybook, Jest and React Testing Library, ESLint and Husky",
}) => (
    <S.Wrapper>
        <S.Logo src="/img/logo.svg" alt="Onyx Logo" />
        <S.Title>{title}</S.Title>
        <S.Description>{description}</S.Description>
    </S.Wrapper>
);

export default Main;
