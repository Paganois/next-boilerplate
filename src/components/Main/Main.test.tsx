import { renderWithTheme } from "utils/tests/helpers";
import { screen } from "@testing-library/react";

import Main from ".";

describe("<Main />", () => {
    it("should render the heading", () => {
        const { container } = renderWithTheme(<Main />);

        expect(
            screen.getByRole("heading", { name: /onyx boilerplate/i })
        ).toBeInTheDocument();

        expect(container.firstChild).toMatchSnapshot();
    });

    it("should render the colors correctly", () => {
        const { container } = renderWithTheme(<Main />);

        expect(container.firstChild).toHaveStyle({
            "background-color": "#1e0b36",
        });
    });
});
