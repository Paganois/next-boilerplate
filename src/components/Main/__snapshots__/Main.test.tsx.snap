// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  background-color: #1e0b36;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 25rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 400;
  margin-top: 1rem;
  width: min(50rem,100%);
}

<main
  class="c0"
>
  <img
    alt="Desenho de um 'O' realacionado ao nome da empresa Onyx"
    class="c1"
    src="/img/logo.svg"
  />
  <h1
    class="c2"
  >
    Onyx Boilerplate
  </h1>
  <h2
    class="c3"
  >
    Boilerplate with Typescript, Next.js, Styled Components, Storybook, Jest and ESLint
  </h2>
</main>
`;
